### ISSUE TITLE
Send Buffer but receive string

### ISSUE THREAD

**Issue by StarpTech** [nats-io/node-nats#108]
Hi, when I send a buffer the payload is converted to a string. https://github.com/nats-io/node-nats/blob/master/lib/nats.js#L851 This is definitly wrong. Encoding should be the job of the developer or be configurable by an additional option. The current solution stop me to use msgpack. I could create a PR.

**Comment by derekcollison** [comment]
https://github.com/nats-io/node-nats/blob/master/test/binary.js#L25

**Comment by aricart** [comment]
You need to set the `encoding` on the client. Then you'll get exactly what you need.

**Comment by StarpTech** [comment]
Setting stream encoding and replace these lines would fix this issue:

```js
806:      // If we are here we have the complete message.
      // Check to see if we have existing chunks
      if (client.payload.chunks) {
	client.payload.chunks.push(client.inbound.slice(0, client.payload.psize));
	var mbuf = Buffer.concat(client.payload.chunks, client.payload.size+CR_LF_LEN);
	client.payload.msg = mbuf
      } else {
        client.payload.msg = new Buffer(client.inbound, 0, client.payload.size, client.encoding);
   
      }
```

**Comment by aricart** [comment]
This basically keeps payload.msg as Buffer. But there are a couple of issues:

- What happens to clients in the wild that currently are getting string?
- It also probably breaks https://github.com/nats-io/node-nats/blob/master/lib/nats.js#L930

I understand that we need to normalize this. But in node-nats-streaming we are sending and receiving protobufs. We handled it in the client by converting the message to a Buffer, with binary encoding `new Buffer(msg, 'binary')`.

I will need to make sure that we have a test proving that existing clients are not affected by this change.

**Comment by StarpTech** [comment]
>What happens to clients in the wild that currently are getting string?

We could release a major release 1.0.0  and normalize all the places.

**Comment by tutman96** [comment]
Could add an option 'preserveBuffers' that ignores the encoding and returns raw buffers. That way it is 100% compatible yet enableable for new clients.

**Comment by StarpTech** [comment]
Sounds good but as soon as the next major release is released I would set it as default. Node-nats is the native driver and shouldnt contain such custom encoding.

**Comment by StarpTech** [comment]
WDYT? @aricart

**Comment by StarpTech** [comment]
As long as the test don't run on windows I can't work on it https://github.com/nats-io/node-nats/issues/109

**Comment by StarpTech** [comment]
PR was created https://github.com/nats-io/node-nats/pull/110

**Comment by aricart** [comment]
Fixed by #111

**Comment by aricart** [comment]
Also this is now available using npm install (0.7.2)

**Comment by StarpTech** [comment]
Cool, thanks!
